source/fuzz/fuzzer.h

```diff
@@ -83,8 +75,13 @@ class Fuzzer {
   // returns a successful result status together with the transformed binary and
   // the sequence of transformations that were applied.  Otherwise, returns an
   // appropriate result status together with an empty binary and empty
-  // transformation sequence.
-  FuzzerResult Run();
+  // transformation sequence. |num_of_transformations| is equal to the maximum
+  // number of transformations applied in a single call to this method. This
+  // parameter is ignored if its value is equal to 0.
+  FuzzerResult Run(uint32_t num_of_transformations_to_apply = 0);
+
+  // Returns all applied transformations.
+  const protobufs::TransformationSequence& GetAppliedTransformations() const;
```

Comment:
It feels a bit wrong to me that the transformed binary is returned as part of FuzzerResult, yet the applied transformations are obtained separately via this method.

I wonder if it would actually be better to:

- Stop "Run" from dumping the module to a binary
- Change the return type of "Run" to just return a status (so get rid of the FuzzerResult struct)
- Have a getter method for the module

That way we put the burden on the client to dump the module to a binary - and that's good because the client might not actually want to do that (e.g., the client might want the IR context so that it can do something further with it).

What do you think?

Reply:
Fixed in 5967f1f.